src/parser/validate.ts

```diff
@@ -57,44 +57,104 @@ export function validate(parse: {
   function validateFlags() {
     for (const [name, flag] of Object.entries(parse.input.flags)) {
       if (parse.output.flags[name] !== undefined) {
-        for (const also of flag.dependsOn || []) {
-          if (!parse.output.flags[also]) {
-            throw new CLIError(
-              `--${also}= must also be provided when using --${name}=`,
-            )
+        validateRelationships(name, flag)
+        validateDependsOn(name, flag.dependsOn ?? [])
+        validateExclusive(name, flag.exclusive ?? [])
+        validateExactlyOne(name, flag.exactlyOne ?? [])
+      } else if (flag.required) {
+        throw new RequiredFlagError({parse, flag})
+      } else if (flag.exactlyOne && flag.exactlyOne.length > 0) {
+        validateAcrossFlags(flag)
+      }
+    }
+  }
+
+  function validateExclusive(name: string, exclusive: string[]) {
+    for (const also of exclusive) {
+      // do not enforce exclusivity for flags that were defaulted
```

Comment:
consider making this configurable.  Flag B exists, but can't be used unless you make Flag A something other than its default